### ISSUE TITLE
Big Number of API requests

### ISSUE THREAD

**Issue by paglias** [PitiTheGrey/Habitican-Wardrobes#1]
Hi @PitiTheGrey , Matteo from Habitica.

I'm opening this issue because we've seen from our server logs that there have been a few instances of users spamming the API through the `Habitican-Wardrobes` making thousands of calls in a short period of time to the cast skill API.

Is it possible to add a check making sure that API calls are limited to 1 every 30s ?

Thanks

**Comment by PitiTheGrey** [comment]
Hi Matteo!  
Thank you for opening this issue! Habitican Wardrobes is meant to help players, rather than spamming the servers.  
I have added a waitTime between skill-calls. The higher the amount of skills is set, the longer the waittime will be. This feature was already implemented long time ago, but it got lost, due to not being called. Fixed that.  

Really sorry for the inconvenience!

**Comment by paglias** [comment]
thanks @PitiTheGrey ! And no worries, it happens, I'm going to close the issue but will let you know if it happens again :)

**Comment by paglias** [comment]
Hi @PitiTheGrey , sorry for bothering again but a few hours ago we got another 6 thousand skills casted by a single user in under 1 hour, could there be an issue with the timer?

**Comment by paglias** [comment]
Actually it was 6k skills casted in 20 minutes

**Comment by PitiTheGrey** [comment]
Hi Matteo!
Sorry for being a trouble here.
I added " Habitican Wardrobes v.2.5 " to the x-api-header, so in future you
can evaluate if it is my current version causing friction, or if it is a
users copy of an old version.
Also I have set a threshold of 20 to the input fields. Input fields were
limited to 20 anyway (with a non-obvious option to enter a higher number of
skills manually). If the skill-button is fired now,
code will limit number to 20 AND give an (annoying) alert.
I really hope this helps!
CU, Piti!

Am Di., 14. Apr. 2020 um 12:43 Uhr schrieb Matteo Pagliazzi <
notifications@github.com>:
